tests/e2e/customsuites/mountoptions.go

```diff
@@ -100,74 +101,197 @@ func (t *s3CSIMountOptionsTestSuite) DefineTests(driver storageframework.TestDri
-		// - uid/gid set to non-root values
-		// - allow-other to permit access by users other than the mounter
-		// - debug flags for better logging in case of issues
-		resource := createVolumeResourceWithMountOptions(ctx, l.config, pattern, []string{
-			fmt.Sprintf("uid=%d", DefaultNonRootUser),
-			fmt.Sprintf("gid=%d", DefaultNonRootGroup),
-			"allow-other",
+		// Use BuildVolumeWithOptions from util.go which provides standard non-root options
+		// plus debug for better diagnostics
+		resource := BuildVolumeWithOptions(
+			ctx,
+			l.config,
+			pattern,
+			DefaultNonRootUser,
+			DefaultNonRootGroup,
+			"", // No specific file mode
 			"debug",
-		})
+		)
 		l.resources = append(l.resources, resource)
+
 		ginkgo.By("Creating pod with a volume")
-		pod := e2epod.MakePod(f.Namespace.Name, nil, []*v1.PersistentVolumeClaim{resource.Pvc}, admissionapi.LevelRestricted, "")
-		podModifierNonRoot(pod)
+		pod := MakeNonRootPodWithVolume(f.Namespace.Name, []*v1.PersistentVolumeClaim{resource.Pvc}, "")
 		var err error
 		pod, err = createPod(ctx, f.ClientSet, f.Namespace.Name, pod)
 		framework.ExpectNoError(err)
 		defer func() {
 			framework.ExpectNoError(e2epod.DeletePodWithWait(ctx, f.ClientSet, pod))
 		}()
+
 		volPath := "/mnt/volume1"
 		fileInVol := fmt.Sprintf("%s/file.txt", volPath)
 		seed := time.Now().UTC().UnixNano()
 		toWrite := 1024 // 1KB
+
 		ginkgo.By("Checking write to a volume")
 		checkWriteToPath(f, pod, fileInVol, toWrite, seed)
+
 		ginkgo.By("Checking read from a volume")
 		checkReadFromPath(f, pod, fileInVol, toWrite, seed)
+
 		ginkgo.By("Checking file group owner")
 		e2evolume.VerifyExecInPodSucceed(f, pod, fmt.Sprintf("stat -L -c '%%a %%g %%u' %s | grep '644 %d %d'", fileInVol, DefaultNonRootGroup, DefaultNonRootUser))
+
 		ginkgo.By("Checking dir group owner")
 		e2evolume.VerifyExecInPodSucceed(f, pod, fmt.Sprintf("stat -L -c '%%a %%g %%u' %s | grep '755 %d %d'", volPath, DefaultNonRootGroup, DefaultNonRootUser))
+
 		ginkgo.By("Checking pod identity")
 		e2evolume.VerifyExecInPodSucceed(f, pod, fmt.Sprintf("id | grep 'uid=%d gid=%d groups=%d'", DefaultNonRootUser, DefaultNonRootGroup, DefaultNonRootGroup))
 	}
 	ginkgo.It("should access volume as a non-root user", func(ctx context.Context) {
 		validateWriteToVolume(ctx)
 	})
 
-	// accessVolAsNonRootUser is a helper function that tests that access is properly denied
-	// when mount options for non-root access are NOT provided.
+	// ---------------------------------------------------------------------------
+	// Unsupported Mount-arg tests
 	//
-	// This function:
-	// 1. Creates a volume with NO special mount options
-	// 2. Creates a pod that runs as non-root and tries to access this volume
-	// 3. Verifies that access is denied due to lack of permissions
+	// Context
+	// -------
+	// If any of these flags reach Mountpoint-S3 it refuses writes or targets the
+	// wrong backend:
 	//
-	// This is security test to ensure volumes aren't accessible
```

Comment:
Security test doesn't really beloing here, we have other tests for this now under customsuites